velox/exec/MergeJoin.h

```diff
@@ -424,18 +427,17 @@ class MergeJoin : public Operator {
       return currentLeftRowNumber_ == rawLeftRowNumbers_[row];
     }
 
-    // Called when all rows from the current output batch are processed and the
-    // next batch of output will start with a new left-side row or there will
-    // be no more batches. Calls 'onMiss' for the last left-side row if the
-    // filter failed for all matches of that row.
+    // Called when all rows from the current output batch are processed. Calls
+    // 'onMiss' for the last left-side row if the filter failed for all matches
+    // of that row.
     template <typename TOnMiss>
     void noMoreFilterResults(TOnMiss onMiss) {
-      if (!currentRowPassed_) {
+      if (!currentRowPassed_ && currentRow_ >= 0) {
```

Comment:
In which use cases would currentRow_ < 0 occur when calling the noMoreFilterResults method?